Create new food category for ASRS.

Huffie56 opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->

## About The Pull Request

1-creating a new category in _supply_packs.dm
2-creating a new file named food.dm
3-add new crates with (potato wheat tomato)*20 meat and fish  *5 and a crate with every type of condiment.
3-populating the files with all the current food crate related items in ASRS.
4-Removing the duplicate crate from misc.dm and operations.dm

<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game
With this new category people and cook will clearly see where to look if they want food items.
<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding, and may discourage maintainers from reviewing or merging your PR. This section is not strictly required for (non-controversial) fix PRs or backend PRs. -->

## Changelog

<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

<!-- If your PR modifies aspects of the game that can be concretely observed by players or admins you should add a changelog. If your change does NOT meet this description, remove this section. Be sure to properly mark your PRs to prevent unnecessary GBP loss. Please note that maintainers freely reserve the right to remove and add tags should they deem it appropriate. You can attempt to finagle the system all you want, but it's best to shoot for clear communication right off the bat. -->

:cl:
add: Added a new category for ASRS : food.
add: new crates with (potato wheat tomato)*20 meat and fish  *5 and a crate with every type of condiment.
qol: regrouping all the food crate related item in the food category
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->

fira commented:
You need to:
 1. Include the new file in the DME
 2. Assign cost to a variable (cost = probably)
 3. Put the rest of the food in that category (eg ingredients crate)
But yeah it's a good idea

Huffie56 commented:
> You need to:
> 
>     1. Include the new file in the DME
> 
>     2. Assign cost to a variable (cost = probably)
> 
>     3. Put the rest of the food in that category (eg ingredients crate)
>        But yeah it's a good idea

thanks : 
1-i didn't saw any dme file linked to ASRS code. can you clarify what file i need to modify? i added the category in supplypack.dm
2-their is a cost assign via cost = RO_PRICE_WORTHLESS variable . not sure what you mean there...
3-I just did remove all the food from misc category  and one from operation.
please help for 1 and 2.

fira commented:
well, most of these are missing the `cost =` part
also the new file mut be inside the main environment file `ColonialMarinesALPHA.dme`
i'd suggest running this through dreammaker and testing it to ensure it works

Huffie56 commented:
> well, most of these are missing the `cost =` part also the new file mut be inside the main environment file `ColonialMarinesALPHA.dme` i'd suggest running this through dreammaker and testing it to ensure it works

I am not sure when i manage to delete cost = . I corrected it.
I added the define in ColonialMarinesALPHA.dme fallowing the alphabetic order like it was.
Do i really need to run and test it via dreammaker? and if yes how?
thanks

fira commented on code/datums/supply_packs/food.dm:
```diff
@@ -0,0 +1,223 @@
+	containername = "\improper USCM MRE crate(x2)"
+	group = "Food"
+	
+	/datum/supply_packs/funfood
```
indentation error here, there's one extra tab

Huffie56 replied:
i see thanks

fira reviewed (changes requested):
The prefered way to test is to download VScode, get the game files, open the folder in VScode, then you can press F5 to build/launch the game and try things out. Don't hesitate to ask on discord if you need help with this

Huffie56 commented:
> The prefered way to test is to download VScode, get the game files, open the folder in VScode, then you can press F5 to build/launch the game and try things out. Don't hesitate to ask on discord if you need help with this

I will try and come to discord if needed.
thanks

Huffie56 commented:
tested it's working
ASRS menu and ordering food

![image](https://user-images.githubusercontent.com/117036822/202023265-339516a3-90dd-4759-9813-1486ad6ae6dd.png)

I changed the long ass description
Req result
![image](https://user-images.githubusercontent.com/117036822/202023023-49ac3b45-922a-4e61-81f3-7e7ff1278efb.png)
